lib/puppet/provider/svckill/kill.rb

```diff
@@ -140,16 +140,21 @@ def mode
         end
       elsif [:redhat,:systemd].include?(res[:provider])
         if res[:enable].to_s.eql?("true") || res[:ensure].eql?(:running)
-          @running_services[obj[:name]] = {
-            :provider => obj.provider,
-            :resource => res
-          }
+          #  Only add services that are enabled or disabled.  Killing static or other services can cause
+          #  system errors.
+          if obj.provider.respond_to?(:cached_enabled?) && ['enabled', 'disabled'].include?(obj.provider.cached_enabled?[:output])
```

Comment:
```suggestion
          if !obj.provider.respond_to?(:cached_enabled?) || (obj.provider.respond_to?(:cached_enabled?) && ['enabled', 'disabled'].include?(obj.provider.cached_enabled?[:output]))
```

This should cover the non-systemd cases. Good call on pruning prior to adding, much more efficient.